### PR TITLE
Fix 2547 - Added fix_javascript() to URLs to prevent XSS

### LINKED ISSUE #2547: XSS in URL BBCode
The URL BBCode is not sanitized properly, leading posts/PMs open to XSS attacks. Internal thread: https://community.mybb.com/thread-206010.html

### PR DESCRIPTION
**Pull request by Shade-**
Fixes #2547. Tested and should work fine. Further testing might be needed @mybb/sqa

### CODE REVIEW

File: inc/class_parser.php [hunk 1]
```diff
@@ -1060,6 +1060,12 @@ function mycode_parse_url($url, $name="")
 		{
 			$nofollow = " rel=\"nofollow\"";
 		}
+		
+		// Fix eventual XSS attempts
+		if(strpos($url, 'javascript:') !== false)
```
Comment (JN-Jones) [inline inc/class_parser.php]: Shouldn't this check for position 0? Otherwise URLs like `http://example.com/javascript:something` might be changed too.

**Comment by Shade-** [comment]
Would it be better to strip out javascript:// occurrencies then?

**Comment by Devilshakerz** [comment]
Either the `[url]` regexes should be changed (it would be the only disallowed protocol) or their callback functions should ignore the part if `^javascript:` is matched.

File: inc/class_parser.php (hunk 1 above)
Comment (Ben-Conway) [inline inc/class_parser.php]: @Shade- ?

**Comment by Ben-Conway** [comment]
Can we get this fixed and merged? @Shade-

**Comment by Shade-** [comment]
Yep sorry Ben, I'm studying at the moment but tomorrow it's vacation time here so I should push the fixes I've already made for this on my localhost.

**Comment by Ben-Conway** [comment]
@Stefan-ST @JN-Jones Are we happy with the new changes?

**Comment by Devilshakerz** [comment]
Submitted https://github.com/mybb/mybb/pull/2559 with more & modified lookaheads to prevent false negatives (`[url]javascript://h4x()[/url]`) and positives (`[url=javascript.com]test[/url]`)

**Pull request closed without merging**